src/Kurrent.Client/Core/Serialization/SystemTextJsonSerializer.cs

```diff
@@ -0,0 +1,36 @@
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using EventStore.Client.Serialization;
+
+namespace Kurrent.Client.Core.Serialization;
+
+public class SystemTextJsonSerializationSettings {
+	public static readonly JsonSerializerOptions DefaultJsonSerializerOptions =
+		new JsonSerializerOptions(JsonSerializerOptions.Default) {
+			PropertyNamingPolicy        = JsonNamingPolicy.CamelCase,
+			DictionaryKeyPolicy         = JsonNamingPolicy.CamelCase,
+			PropertyNameCaseInsensitive = false,
+			DefaultIgnoreCondition      = JsonIgnoreCondition.WhenWritingNull,
+			UnknownTypeHandling         = JsonUnknownTypeHandling.JsonNode,
+			UnmappedMemberHandling      = JsonUnmappedMemberHandling.Skip,
+			NumberHandling              = JsonNumberHandling.AllowReadingFromString,
+			Converters = {
+				new JsonStringEnumConverter(JsonNamingPolicy.CamelCase),
+			}
+		};
+
+	public JsonSerializerOptions Options { get; set; } = DefaultJsonSerializerOptions;
+}
+
+public class SystemTextJsonSerializer(SystemTextJsonSerializationSettings? options = null) : ISerializer {
+	readonly JsonSerializerOptions _options = options?.Options ?? SystemTextJsonSerializationSettings.DefaultJsonSerializerOptions;
+
+	public ReadOnlyMemory<byte> Serialize(object value) {
+		return Encoding.UTF8.GetBytes(JsonSerializer.Serialize(value, _options));
```

Comment:
Why not use SerializeToUtf8Bytes?

Reply:
Fixed in https://github.com/EventStore/EventStore-Client-Dotnet/pull/329/commits/e8385b38d24f5eb138b8393a4acbcdc0c6416a9b